speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp

```diff
@@ -18,9 +18,32 @@
         ASSERT_TRUE(file.is_open());                                                                \
         std::stringstream buffer;                                                                   \
         buffer << file.rdbuf();                                                                     \
+        std::cout << buffer.str() << std::endl;                                                     \
         ASSERT_STREQ(buffer.str().c_str(), contents);                                               \
     }
 
+/// @brief Assert the contents of [filename] of recording [recordingName] located in [folder] match the binary [contents]
+#define ASSERT_BINARY_FILE_CONTENTS(folder, recordingName, filename, contents)                             \
+    {                                                                                                      \
+        std::cout << "Reading binary file from " << (folder + (recordingName "\\" filename)) << std::endl; \
+        std::ifstream file(folder + ("\\" recordingName "\\" filename));                                   \
+        ASSERT_TRUE(file.is_open());                                                                       \
+        std::stringstream buffer;                                                                          \
+        buffer << file.rdbuf();                                                                            \
+        file.close();                                                                                      \
+        ASSERT_TRUE(!file.is_open());                                                                      \
+        ASSERT_EQ(buffer.str().size(), contents.str().size());                                             \
+        for (int i = 0; i < buffer.str().size(); i++)                                                      \
+        {                                                                                                  \
+            char controlByte;                                                                              \
+            char testByte;                                                                                 \
+            contents >> bits(controlByte);                                                                 \
+            buffer >> bits(testByte);                                                                      \
+            std::cout << "Checking... " << testByte << " == " << controlByte << std::endl;                 \
```

Comment:
No need to log that every byte, causes a lot of spam in the logs and isn't useful since most bytes aren't printable.

Reply:
Right, that was for debugging, I'll delete it